pkg/machine/ignition.go

```diff
@@ -365,52 +383,111 @@ machine_enabled=true
+				setSSLCertFile := fmt.Sprintf("export %s=%s", "SSL_CERT_FILE", filepath.Join("/etc/containers/certs.d", filepath.Base(sslCertFile)))
+				files = append(files, File{
+					Node: Node{
+						Group: getNodeGrp("root"),
+						Path:  "/etc/profile.d/ssl_cert_file.sh",
+						User:  getNodeUsr("root"),
+					},
+					FileEmbedded1: FileEmbedded1{
+						Append: nil,
+						Contents: Resource{
+							Source: encodeDataURLPtr(setSSLCertFile),
+						},
+						Mode: intToPtr(0644),
+					},
+				})
+			}
+		}
+	}
+
 	return files
 }
 
-func getCerts(certsDir string) []File {
+func getCerts(certsDir string, isDir bool) []File {
 	var (
 		files []File
 	)
 
 	certs, err := ioutil.ReadDir(certsDir)
-	if err == nil {
-		for _, cert := range certs {
-			b, err := ioutil.ReadFile(filepath.Join(certsDir, cert.Name()))
-			if err != nil {
-				logrus.Warnf("Unable to read cert file %s", err.Error())
-				continue
-			}
-			files = append(files, File{
-				Node: Node{
-					Group: getNodeGrp("root"),
-					Path:  filepath.Join("/etc/containers/certs.d/", cert.Name()),
-					User:  getNodeUsr("root"),
-				},
-				FileEmbedded1: FileEmbedded1{
-					Append: nil,
-					Contents: Resource{
-						Source: encodeDataURLPtr(string(b)),
+	if isDir {
+		if err == nil {
+			for _, cert := range certs {
+				b, err := ioutil.ReadFile(filepath.Join(certsDir, cert.Name()))
+				if err != nil {
+					logrus.Warnf("Unable to read cert file %s", err.Error())
+					continue
+				}
+				files = append(files, File{
+					Node: Node{
+						Group: getNodeGrp("root"),
+						Path:  filepath.Join("/etc/containers/certs.d/", cert.Name()),
+						User:  getNodeUsr("root"),
 					},
-					Mode: intToPtr(0644),
-				},
-			})
+					FileEmbedded1: FileEmbedded1{
+						Append: nil,
+						Contents: Resource{
+							Source: encodeDataURLPtr(string(b)),
+						},
+						Mode: intToPtr(0644),
+					},
+				})
+			}
+		} else {
+			if !os.IsNotExist(err) {
+				logrus.Warnf("Unable to copy certs via ignition, error while reading certs from %s:  %s", certsDir, err.Error())
+			}
 		}
 	} else {
-		if !os.IsNotExist(err) {
-			logrus.Warnf("Unable to copy certs via ignition, error while reading certs from %s:  %s", certsDir, err.Error())
+		fileName := filepath.Base(certsDir)
+		b, err := ioutil.ReadFile(certsDir)
+		if err != nil {
+			logrus.Warnf("Unable to read cert file %s", err.Error())
+			return files
 		}
+		files = append(files, File{
+			Node: Node{
+				Group: getNodeGrp("root"),
+				Path:  filepath.Join("/etc/containers/certs.d/", fileName),
+				User:  getNodeUsr("root"),
+			},
+			FileEmbedded1: FileEmbedded1{
+				Append: nil,
+				Contents: Resource{
+					Source: encodeDataURLPtr(string(b)),
+				},
+				Mode: intToPtr(0644),
+			},
+		})
 	}
 
 	return files
 }
 
+func getProxyVariables() string {
+	proxyOpts := ""
+	proxyVariables := []string{"HTTP_PROXY", "HTTPS_PROXY", "NO_PROXY", "http_proxy", "https_proxy", "no_proxy", "HELLO", "WORLD"}
```

Comment:
HELLO and WORLD are not Proxy Variables as far as I know.?

Reply:
There is also a list here.
./pkg/specgen/generate/container.go

Can you put these into a shred location. and use them in both places.

Reply:
@rhatdan I was testing something forgot to remove that i guess :sweat_smile:

Reply:
What about https_proxy and ftp_proxy?

Reply:
@rhatdan I moved them to common place in `pkg/utils` does that looks good to you.